src/initialization/InitElement.cpp

```diff
@@ -179,6 +179,12 @@ namespace detail
                 detail::queryAddResize(pp_element, "cos_coefficients", cos_coef);
                 detail::queryAddResize(pp_element, "sin_coefficients", sin_coef);
                 m_lattice.emplace_back( SoftQuadrupole(ds, gscale, cos_coef, sin_coef, mapsteps, nslice) );
+            } else if (element_type == "beam_monitor") {
+                std::string openpmd_backend = "default";
+                pp_element.queryAdd("backend", openpmd_backend);
+                std::string  openpmd_encoding {"g"};
+                pp_element.queryAdd("encoding", openpmd_encoding);
+                m_lattice.emplace_back( diagnostics::BeamMonitor(element_name, openpmd_backend, openpmd_encoding) );
```

Comment:
I have two questions about this element.  First, is it useful to include an ascii option for backward-compatibility?  (Some users are simply more comfortable with plain ascii text.)  Second, should we include the option of specifying a filename for output?

Reply:
I think no and yes :-) No to the first part because it's not scalable and what counts for users is mainly an API to read the data in their scripting language of choice.

Let's discuss tomorrow :)